R/calcACOwnershipRates.R

```diff
@@ -0,0 +1,166 @@
+  odysseeData <- readSource("Odyssee") %>%
+    as.quitte(na.rm = TRUE)
+
+  # population (aggregation weights)
+  pop <- calcOutput("Population", scenario = "SSP2", aggregate = FALSE) %>%
+    as.quitte() %>%
+    removeColNa()
+
+
+  ## Mappings ====
+
+  regionmapIEA <- toolGetMapping("regionmappingIEA_acOwnership.csv",
+                                 type = "regional",
+                                 where = "mredgebuildings")
+
+  regionmapEDGE <- toolGetMapping("regionmappingISO-EDGE_EUR_ETP.csv",
+                                  type = "regional",
+                                  where = "mredgebuildings")
+
+
+
+  # PROCESS DATA ---------------------------------------------------------------
+
+  ## Process individual data sets ====
+
+  ### Household surveys ====
+
+  surveyData <- surveyData %>%
+    filter(!is.na(.data$value),
+           .data$variable == "ac") %>%
+    select(-"variable") %>%
+    rename("valueSurvey" = "value")
+
+
+  ### IEA ====
+
+  ieaData <- ieaData %>%
+    filter(!is.na(.data$value))
+
+  # filter out single country data points and give them priority
+  ieaSingleCountry <- ieaData %>%
+    filter(!.data$region %in% regionmapIEA$regionIEA) %>%
+    select(-"variable") %>%
+    rename("valueIEASingle" = "value") %>%
+    mutate(region = as.character(.data$region))
+
+  ieaSingleCountry$region <- toolCountry2isocode(ieaSingleCountry$region)
+
+  #TODO: filter regional values and disaggregate them to fill missing values? #nolint
+
+
+  ### Odyssee ====
+
+  # filter "AC equipment rate" and take most recent data point per region where necessary
+  odysseeData <- odysseeData %>%
+    filter(.data$variable == "teqcli") %>%
+    select("region", "period", "value") %>%
+    left_join(regionmapEDGE, by = c("region" = "CountryCode")) %>%
+    group_by(.data$region) %>%
+    (\(x) {
+      # split data into (non-)single-country regions
+      nonSingleCountry <- filter(x, .data$region != .data$RegionCodeEUR_ETP)
+      singleCountry    <- filter(x, .data$region == .data$RegionCodeEUR_ETP)
+      # take only max period for non-single-country regions
+      nonSingleCountry <- nonSingleCountry %>%
+        slice_max(order_by = .data$period, n = 1, with_ties = FALSE)
+      bind_rows(nonSingleCountry, singleCountry)
+    })() %>%
+    ungroup() %>%
+    rename("valueOdyssee" = "value") %>%
+
+    # Montenegro (MNE) shows an ownership rate of 100% which seems unrealistic
+    filter(.data$region != "MNE")
+
+
+  ## Merge Datasets ====
+
+  # target : obtain merged data set with one data point per region
+  # merge priority : odyssee > iea > surveys
+
+  ownershipRates <- odysseeData %>%
+    full_join(ieaSingleCountry %>%
+                anti_join(odysseeData, by = "region"),
+              by = c("region", "period")) %>%
+    full_join(surveyData %>%
+                anti_join(odysseeData, by = "region") %>%
+                anti_join(ieaSingleCountry, by = "region"),
+              by = c("region", "period")) %>%
```

Comment:
Why do you need to remove rows with `anti_join`? Shouldn't `coalesce` do the hierarchical selection in any case?
```suggestion
    full_join(ieaSingleCountry, by = c("region", "period")) %>%
    full_join(surveyData, by = c("region", "period")) %>%
```